rsbinder-aidl: Incorrect path resolution for cross-package enum default values

When a parcelable references an enum from a different package and uses a default value, the generator incorrectly generates super::SubEnum::SubEnum::UNKNOWN in the Default impl. Since SubEnum is in the test.sub module (not test), the super:: path doesn't resolve correctly, causing a compilation error:

```
error[E0433]: failed to resolve: could not find `SubEnum` in `super`
    r#enumField: super::SubEnum::SubEnum::UNKNOWN,
                        ^^^^^^^ could not find `SubEnum` in `super`
```

Files to reproduce this issue:
```
package test.sub;

@VintfStability
@Backing(type="int")
enum SubEnum {
    UNKNOWN = 0,
}
```

```
package test;

import test.sub.SubEnum;

@VintfStability
parcelable CrossPackageParcelable {
    SubEnum enumField = SubEnum.UNKNOWN;
}
```

Expected behavior: The generator should produce the correct fully-qualified path to the enum, e.g., super::super::sub::SubEnum::SubEnum::UNKNOWN or use a proper import.

--- a/rsbinder-aidl/src/const_expr.rs
+++ b/rsbinder-aidl/src/const_expr.rs
@@ -431,9 +431,16 @@ impl ValueType {
                     // For constants, always use numeric values
                     format!("{}", value)
                 } else {
-                    // For default values, format depends on whether target expects enum or numeric type
-                    // This will be handled in the init_value method based on target type
-                    format!("super::{}::{}::{}", enum_name, enum_name, member_name)
+                    // Use proper namespace resolution for cross-package enum references
+                    let lookup_decl =
+                        parser::lookup_decl_from_name(enum_name, crate::Namespace::AIDL);
+                    let curr_ns = parser::current_namespace();
+                    let ns = curr_ns.relative_mod(&lookup_decl.ns);
+                    if !ns.is_empty() {
+                        format!("{}::{}::{}", ns, enum_name, member_name)
+                    } else {
+                        format!("{}::{}", enum_name, member_name)
+                    }
                 }
             }
             ValueType::Array(v) => {

--- a/rsbinder-aidl/src/generator.rs
+++ b/rsbinder-aidl/src/generator.rs
@@ -690,7 +690,7 @@ impl Generator {
             }
         }
 
-        let enabled_async = self.enabled_async || cfg!(feature = "async");
+        let enabled_async = self.enabled_async;
 
         let nested = &self.declarations(&decl.members, indent + 1)?;
 

--- a/rsbinder-aidl/src/lib.rs
+++ b/rsbinder-aidl/src/lib.rs
@@ -115,7 +115,7 @@ impl Builder {
             includes: Vec::new(),
             dest_dir: PathBuf::from(std::env::var_os("OUT_DIR").unwrap_or("aidl_gen".into())),
             output: "rsbinder_generated_aidl.rs".into(),
-            enabled_async: false,
+            enabled_async: cfg!(feature = "async"),
             is_crate: false,
         }
     }

--- a/rsbinder-aidl/tests/test_arrays.rs
+++ b/rsbinder-aidl/tests/test_arrays.rs
@@ -49,66 +49,6 @@ pub mod ITestService {
             DEFAULT_IMPL.get_or_init(|| d).clone()
         }
     }
-    pub trait ITestServiceAsync<P>: rsbinder::Interface + Send {
-        fn descriptor() -> &'static str where Self: Sized { "android.aidl.fixedsizearray.ITestService" }
-        fn r#ReverseBoolean<'a>(&'a self, _arg_input: &'a [bool], _arg_repeated: &'a mut Vec<bool>) -> rsbinder::BoxFuture<'a, rsbinder::status::Result<Vec<bool>>>;
-        fn r#RepeatNullableIntArray<'a>(&'a self, _arg_input: Option<&'a [i32]>) -> rsbinder::BoxFuture<'a, rsbinder::status::Result<Option<Vec<i32>>>>;
-        fn r#FillOutStructuredParcelable<'a>(&'a self, _arg_parcel: &'a mut rsbinder::Strong<dyn StructuredParcelable>) -> rsbinder::BoxFuture<'a, rsbinder::status::Result<()>>;
-    }
-    #[::async_trait::async_trait]
-    pub trait ITestServiceAsyncService: rsbinder::Interface + Send {
-        fn descriptor() -> &'static str where Self: Sized { "android.aidl.fixedsizearray.ITestService" }
-        async fn r#ReverseBoolean(&self, _arg_input: &[bool], _arg_repeated: &mut Vec<bool>) -> rsbinder::status::Result<Vec<bool>>;
-        async fn r#RepeatNullableIntArray(&self, _arg_input: Option<&[i32]>) -> rsbinder::status::Result<Option<Vec<i32>>>;
-        async fn r#FillOutStructuredParcelable(&self, _arg_parcel: &mut rsbinder::Strong<dyn StructuredParcelable>) -> rsbinder::status::Result<()>;
-    }
-    impl BnTestService
-    {
-        pub fn new_async_binder<T, R>(inner: T, rt: R) -> rsbinder::Strong<dyn ITestService>
-        where
-            T: ITestServiceAsyncService + Sync + Send + 'static,
-            R: rsbinder::BinderAsyncRuntime + Send + Sync + 'static,
-        {
-            struct Wrapper<T, R> {
-                _inner: T,
-                _rt: R,
-            }
-            impl<T, R> rsbinder::Interface for Wrapper<T, R> where T: rsbinder::Interface, R: Send + Sync {
-                fn as_binder(&self) -> rsbinder::SIBinder { self._inner.as_binder() }
-                fn dump(&self, _writer: &mut dyn std::io::Write, _args: &[String]) -> rsbinder::Result<()> { self._inner.dump(_writer, _args) }
-            }
-            impl<T, R> BnTestServiceAdapter for Wrapper<T, R>
-            where
-                T: ITestServiceAsyncService + Sync + Send + 'static,
-                R: rsbinder::BinderAsyncRuntime + Send + Sync + 'static,
-            {
-                fn as_sync(&self) -> &dyn ITestService {
-                    self
-                }
-                fn as_async(&self) -> &dyn ITestServiceAsyncService {
-                    &self._inner
-                }
-            }
-            impl<T, R> ITestService for Wrapper<T, R>
-            where
-                T: ITestServiceAsyncService + Sync + Send + 'static,
-                R: rsbinder::BinderAsyncRuntime + Send + Sync + 'static,
-            {
-                fn r#ReverseBoolean(&self, _arg_input: &[bool], _arg_repeated: &mut Vec<bool>) -> rsbinder::status::Result<Vec<bool>> {
-                    self._rt.block_on(self._inner.r#ReverseBoolean(_arg_input, _arg_repeated))
-                }
-                fn r#RepeatNullableIntArray(&self, _arg_input: Option<&[i32]>) -> rsbinder::status::Result<Option<Vec<i32>>> {
-                    self._rt.block_on(self._inner.r#RepeatNullableIntArray(_arg_input))
-                }
-                fn r#FillOutStructuredParcelable(&self, _arg_parcel: &mut rsbinder::Strong<dyn StructuredParcelable>) -> rsbinder::status::Result<()> {
-                    self._rt.block_on(self._inner.r#FillOutStructuredParcelable(_arg_parcel))
-                }
-            }
-            let wrapped = Wrapper { _inner: inner, _rt: rt };
-            let binder = rsbinder::native::Binder::new_with_stability(BnTestService(Box::new(wrapped)), rsbinder::Stability::default());
-            rsbinder::Strong::new(Box::new(binder))
-        }
-    }
     pub trait ITestServiceDefault: Send + Sync {
         fn r#ReverseBoolean(&self, _arg_input: &[bool], _arg_repeated: &mut Vec<bool>) -> rsbinder::status::Result<Vec<bool>> {
             Err(rsbinder::StatusCode::UnknownTransaction.into())
@@ -131,11 +71,8 @@ pub mod ITestService {
         ITestService["android.aidl.fixedsizearray.ITestService"] {
             native: {
                 BnTestService(on_transact),
-                adapter: BnTestServiceAdapter,
-                r#async: ITestServiceAsyncService,
             },
             proxy: BpTestService,
-            r#async: ITestServiceAsync,
         }
     }
     impl BpTestService {
@@ -210,68 +147,15 @@ pub mod ITestService {
             self.read_response_FillOutStructuredParcelable(_arg_parcel, _aidl_reply)
         }
     }
-    impl<P: rsbinder::BinderAsyncPool> ITestServiceAsync<P> for BpTestService {
-        fn r#ReverseBoolean<'a>(&'a self, _arg_input: &'a [bool], _arg_repeated: &'a mut Vec<bool>) -> rsbinder::BoxFuture<'a, rsbinder::status::Result<Vec<bool>>> {
-            let _aidl_data = match self.build_parcel_ReverseBoolean(_arg_input, _arg_repeated) {
-                Ok(_aidl_data) => _aidl_data,
-                Err(err) => return Box::pin(std::future::ready(Err(err.into()))),
-            };
-            let binder = self.binder.clone();
-            P::spawn(
-                move || binder.as_proxy().unwrap().submit_transact(transactions::r#ReverseBoolean, &_aidl_data, rsbinder::FLAG_CLEAR_BUF | rsbinder::FLAG_PRIVATE_LOCAL),
-                move |_aidl_reply| async move {
-                    self.read_response_ReverseBoolean(_arg_input, _arg_repeated, _aidl_reply)
-                }
-            )
-        }
-        fn r#RepeatNullableIntArray<'a>(&'a self, _arg_input: Option<&'a [i32]>) -> rsbinder::BoxFuture<'a, rsbinder::status::Result<Option<Vec<i32>>>> {
-            let _aidl_data = match self.build_parcel_RepeatNullableIntArray(_arg_input) {
-                Ok(_aidl_data) => _aidl_data,
-                Err(err) => return Box::pin(std::future::ready(Err(err.into()))),
-            };
-            let binder = self.binder.clone();
-            P::spawn(
-                move || binder.as_proxy().unwrap().submit_transact(transactions::r#RepeatNullableIntArray, &_aidl_data, rsbinder::FLAG_CLEAR_BUF | rsbinder::FLAG_PRIVATE_LOCAL),
-                move |_aidl_reply| async move {
-                    self.read_response_RepeatNullableIntArray(_arg_input, _aidl_reply)
-                }
-            )
-        }
-        fn r#FillOutStructuredParcelable<'a>(&'a self, _arg_parcel: &'a mut rsbinder::Strong<dyn StructuredParcelable>) -> rsbinder::BoxFuture<'a, rsbinder::status::Result<()>> {
-            let _aidl_data = match self.build_parcel_FillOutStructuredParcelable(_arg_parcel) {
-                Ok(_aidl_data) => _aidl_data,
-                Err(err) => return Box::pin(std::future::ready(Err(err.into()))),
-            };
-            let binder = self.binder.clone();
-            P::spawn(
-                move || binder.as_proxy().unwrap().submit_transact(transactions::r#FillOutStructuredParcelable, &_aidl_data, rsbinder::FLAG_CLEAR_BUF | rsbinder::FLAG_PRIVATE_LOCAL),
-                move |_aidl_reply| async move {
-                    self.read_response_FillOutStructuredParcelable(_arg_parcel, _aidl_reply)
-                }
-            )
-        }
-    }
-    impl<P: rsbinder::BinderAsyncPool> ITestServiceAsync<P> for rsbinder::Binder<BnTestService>
-    {
-        fn r#ReverseBoolean<'a>(&'a self, _arg_input: &'a [bool], _arg_repeated: &'a mut Vec<bool>) -> rsbinder::BoxFuture<'a, rsbinder::status::Result<Vec<bool>>> {
-            self.0.as_async().r#ReverseBoolean(_arg_input, _arg_repeated)
-        }
-        fn r#RepeatNullableIntArray<'a>(&'a self, _arg_input: Option<&'a [i32]>) -> rsbinder::BoxFuture<'a, rsbinder::status::Result<Option<Vec<i32>>>> {
-            self.0.as_async().r#RepeatNullableIntArray(_arg_input)
-        }
-        fn r#FillOutStructuredParcelable<'a>(&'a self, _arg_parcel: &'a mut rsbinder::Strong<dyn StructuredParcelable>) -> rsbinder::BoxFuture<'a, rsbinder::status::Result<()>> {
-            self.0.as_async().r#FillOutStructuredParcelable(_arg_parcel)
-        }
-    }
     impl ITestService for rsbinder::Binder<BnTestService> {
         fn r#ReverseBoolean(&self, _arg_input: &[bool], _arg_repeated: &mut Vec<bool>) -> rsbinder::status::Result<Vec<bool>> {
-            self.0.as_sync().r#ReverseBoolean(_arg_input, _arg_repeated)
+            self.0.r#ReverseBoolean(_arg_input, _arg_repeated)
         }
         fn r#RepeatNullableIntArray(&self, _arg_input: Option<&[i32]>) -> rsbinder::status::Result<Option<Vec<i32>>> {
-            self.0.as_sync().r#RepeatNullableIntArray(_arg_input)
+            self.0.r#RepeatNullableIntArray(_arg_input)
         }
         fn r#FillOutStructuredParcelable(&self, _arg_parcel: &mut rsbinder::Strong<dyn StructuredParcelable>) -> rsbinder::status::Result<()> {
-            self.0.as_sync().r#FillOutStructuredParcelable(_arg_parcel)
+            self.0.r#FillOutStructuredParcelable(_arg_parcel)
         }
     }
     fn on_transact(
@@ -416,56 +300,6 @@ pub mod FixedSizeArrayExample {
                 DEFAULT_IMPL.get_or_init(|| d).clone()
             }
         }
-        pub trait IRepeatFixedSizeArrayAsync<P>: rsbinder::Interface + Send {
-            fn descriptor() -> &'static str where Self: Sized { "android.aidl.fixedsizearray.FixedSizeArrayExample.IRepeatFixedSizeArray" }
-            fn r#Repeat2dParcelables<'a>(&'a self, _arg_input: &'a [[super::IntParcelable::IntParcelable; 3]; 2], _arg_repeated: &'a mut [[super::IntParcelable::IntParcelable; 3]; 2]) -> rsbinder::BoxFuture<'a, rsbinder::status::Result<[[super::IntParcelable::IntParcelable; 3]; 2]>>;
-        }
-        #[::async_trait::async_trait]
-        pub trait IRepeatFixedSizeArrayAsyncService: rsbinder::Interface + Send {
-            fn descriptor() -> &'static str where Self: Sized { "android.aidl.fixedsizearray.FixedSizeArrayExample.IRepeatFixedSizeArray" }
-            async fn r#Repeat2dParcelables(&self, _arg_input: &[[super::IntParcelable::IntParcelable; 3]; 2], _arg_repeated: &mut [[super::IntParcelable::IntParcelable; 3]; 2]) -> rsbinder::status::Result<[[super::IntParcelable::IntParcelable; 3]; 2]>;
-        }
-        impl BnRepeatFixedSizeArray
-        {
-            pub fn new_async_binder<T, R>(inner: T, rt: R) -> rsbinder::Strong<dyn IRepeatFixedSizeArray>
-            where
-                T: IRepeatFixedSizeArrayAsyncService + Sync + Send + 'static,
-                R: rsbinder::BinderAsyncRuntime + Send + Sync + 'static,
-            {
-                struct Wrapper<T, R> {
-                    _inner: T,
-                    _rt: R,
-                }
-                impl<T, R> rsbinder::Interface for Wrapper<T, R> where T: rsbinder::Interface, R: Send + Sync {
-                    fn as_binder(&self) -> rsbinder::SIBinder { self._inner.as_binder() }
-                    fn dump(&self, _writer: &mut dyn std::io::Write, _args: &[String]) -> rsbinder::Result<()> { self._inner.dump(_writer, _args) }
-                }
-                impl<T, R> BnRepeatFixedSizeArrayAdapter for Wrapper<T, R>
-                where
-                    T: IRepeatFixedSizeArrayAsyncService + Sync + Send + 'static,
-                    R: rsbinder::BinderAsyncRuntime + Send + Sync + 'static,
-                {
-                    fn as_sync(&self) -> &dyn IRepeatFixedSizeArray {
-                        self
-                    }
-                    fn as_async(&self) -> &dyn IRepeatFixedSizeArrayAsyncService {
-                        &self._inner
-                    }
-                }
-                impl<T, R> IRepeatFixedSizeArray for Wrapper<T, R>
-                where
-                    T: IRepeatFixedSizeArrayAsyncService + Sync + Send + 'static,
-                    R: rsbinder::BinderAsyncRuntime + Send + Sync + 'static,
-                {
-                    fn r#Repeat2dParcelables(&self, _arg_input: &[[super::IntParcelable::IntParcelable; 3]; 2], _arg_repeated: &mut [[super::IntParcelable::IntParcelable; 3]; 2]) -> rsbinder::status::Result<[[super::IntParcelable::IntParcelable; 3]; 2]> {
-                        self._rt.block_on(self._inner.r#Repeat2dParcelables(_arg_input, _arg_repeated))
-                    }
-                }
-                let wrapped = Wrapper { _inner: inner, _rt: rt };
-                let binder = rsbinder::native::Binder::new_with_stability(BnRepeatFixedSizeArray(Box::new(wrapped)), rsbinder::Stability::default());
-                rsbinder::Strong::new(Box::new(binder))
-            }
-        }
         pub trait IRepeatFixedSizeArrayDefault: Send + Sync {
             fn r#Repeat2dParcelables(&self, _arg_input: &[[super::IntParcelable::IntParcelable; 3]; 2], _arg_repeated: &mut [[super::IntParcelable::IntParcelable; 3]; 2]) -> rsbinder::status::Result<[[super::IntParcelable::IntParcelable; 3]; 2]> {
                 Err(rsbinder::StatusCode::UnknownTransaction.into())
@@ -480,11 +314,8 @@ pub mod FixedSizeArrayExample {
             IRepeatFixedSizeArray["android.aidl.fixedsizearray.FixedSizeArrayExample.IRepeatFixedSizeArray"] {
                 native: {
                     BnRepeatFixedSizeArray(on_transact),
-                    adapter: BnRepeatFixedSizeArrayAdapter,
-                    r#async: IRepeatFixedSizeArrayAsyncService,
                 },
                 proxy: BpRepeatFixedSizeArray,
-                r#async: IRepeatFixedSizeArrayAsync,
             }
         }
         impl BpRepeatFixedSizeArray {
@@ -514,30 +345,9 @@ pub mod FixedSizeArrayExample {
                 self.read_response_Repeat2dParcelables(_arg_input, _arg_repeated, _aidl_reply)
             }
         }
-        impl<P: rsbinder::BinderAsyncPool> IRepeatFixedSizeArrayAsync<P> for BpRepeatFixedSizeArray {
-            fn r#Repeat2dParcelables<'a>(&'a self, _arg_input: &'a [[super::IntParcelable::IntParcelable; 3]; 2], _arg_repeated: &'a mut [[super::IntParcelable::IntParcelable; 3]; 2]) -> rsbinder::BoxFuture<'a, rsbinder::status::Result<[[super::IntParcelable::IntParcelable; 3]; 2]>> {
-                let _aidl_data = match self.build_parcel_Repeat2dParcelables(_arg_input, _arg_repeated) {
-                    Ok(_aidl_data) => _aidl_data,
-                    Err(err) => return Box::pin(std::future::ready(Err(err.into()))),
-                };
-                let binder = self.binder.clone();
-                P::spawn(
-                    move || binder.as_proxy().unwrap().submit_transact(transactions::r#Repeat2dParcelables, &_aidl_data, rsbinder::FLAG_CLEAR_BUF | rsbinder::FLAG_PRIVATE_LOCAL),
-                    move |_aidl_reply| async move {
-                        self.read_response_Repeat2dParcelables(_arg_input, _arg_repeated, _aidl_reply)
-                    }
-                )
-            }
-        }
-        impl<P: rsbinder::BinderAsyncPool> IRepeatFixedSizeArrayAsync<P> for rsbinder::Binder<BnRepeatFixedSizeArray>
-        {
-            fn r#Repeat2dParcelables<'a>(&'a self, _arg_input: &'a [[super::IntParcelable::IntParcelable; 3]; 2], _arg_repeated: &'a mut [[super::IntParcelable::IntParcelable; 3]; 2]) -> rsbinder::BoxFuture<'a, rsbinder::status::Result<[[super::IntParcelable::IntParcelable; 3]; 2]>> {
-                self.0.as_async().r#Repeat2dParcelables(_arg_input, _arg_repeated)
-            }
-        }
         impl IRepeatFixedSizeArray for rsbinder::Binder<BnRepeatFixedSizeArray> {
             fn r#Repeat2dParcelables(&self, _arg_input: &[[super::IntParcelable::IntParcelable; 3]; 2], _arg_repeated: &mut [[super::IntParcelable::IntParcelable; 3]; 2]) -> rsbinder::status::Result<[[super::IntParcelable::IntParcelable; 3]; 2]> {
-                self.0.as_sync().r#Repeat2dParcelables(_arg_input, _arg_repeated)
+                self.0.r#Repeat2dParcelables(_arg_input, _arg_repeated)
             }
         }
         fn on_transact(
@@ -616,51 +426,6 @@ pub mod FixedSizeArrayExample {
                 DEFAULT_IMPL.get_or_init(|| d).clone()
             }
         }
-        pub trait IEmptyInterfaceAsync<P>: rsbinder::Interface + Send {
-            fn descriptor() -> &'static str where Self: Sized { "android.aidl.fixedsizearray.FixedSizeArrayExample.IEmptyInterface" }
-        }
-        #[::async_trait::async_trait]
-        pub trait IEmptyInterfaceAsyncService: rsbinder::Interface + Send {
-            fn descriptor() -> &'static str where Self: Sized { "android.aidl.fixedsizearray.FixedSizeArrayExample.IEmptyInterface" }
-        }
-        impl BnEmptyInterface
-        {
-            pub fn new_async_binder<T, R>(inner: T, rt: R) -> rsbinder::Strong<dyn IEmptyInterface>
-            where
-                T: IEmptyInterfaceAsyncService + Sync + Send + 'static,
-                R: rsbinder::BinderAsyncRuntime + Send + Sync + 'static,
-            {
-                struct Wrapper<T, R> {
-                    _inner: T,
-                    _rt: R,
-                }
-                impl<T, R> rsbinder::Interface for Wrapper<T, R> where T: rsbinder::Interface, R: Send + Sync {
-                    fn as_binder(&self) -> rsbinder::SIBinder { self._inner.as_binder() }
-                    fn dump(&self, _writer: &mut dyn std::io::Write, _args: &[String]) -> rsbinder::Result<()> { self._inner.dump(_writer, _args) }
-                }
-                impl<T, R> BnEmptyInterfaceAdapter for Wrapper<T, R>
-                where
-                    T: IEmptyInterfaceAsyncService + Sync + Send + 'static,
-                    R: rsbinder::BinderAsyncRuntime + Send + Sync + 'static,
-                {
-                    fn as_sync(&self) -> &dyn IEmptyInterface {
-                        self
-                    }
-                    fn as_async(&self) -> &dyn IEmptyInterfaceAsyncService {
-                        &self._inner
-                    }
-                }
-                impl<T, R> IEmptyInterface for Wrapper<T, R>
-                where
-                    T: IEmptyInterfaceAsyncService + Sync + Send + 'static,
-                    R: rsbinder::BinderAsyncRuntime + Send + Sync + 'static,
-                {
-                }
-                let wrapped = Wrapper { _inner: inner, _rt: rt };
-                let binder = rsbinder::native::Binder::new_with_stability(BnEmptyInterface(Box::new(wrapped)), rsbinder::Stability::default());
-                rsbinder::Strong::new(Box::new(binder))
-            }
-        }
         pub trait IEmptyInterfaceDefault: Send + Sync {
         }
         pub(crate) mod transactions {
@@ -671,21 +436,13 @@ pub mod FixedSizeArrayExample {
             IEmptyInterface["android.aidl.fixedsizearray.FixedSizeArrayExample.IEmptyInterface"] {
                 native: {
                     BnEmptyInterface(on_transact),
-                    adapter: BnEmptyInterfaceAdapter,
-                    r#async: IEmptyInterfaceAsyncService,
                 },
                 proxy: BpEmptyInterface,
-                r#async: IEmptyInterfaceAsync,
             }
         }
         impl BpEmptyInterface {
         }
         impl IEmptyInterface for BpEmptyInterface {
-        }
-        impl<P: rsbinder::BinderAsyncPool> IEmptyInterfaceAsync<P> for BpEmptyInterface {
-        }
-        impl<P: rsbinder::BinderAsyncPool> IEmptyInterfaceAsync<P> for rsbinder::Binder<BnEmptyInterface>
-        {
         }
         impl IEmptyInterface for rsbinder::Binder<BnEmptyInterface> {
         }

--- a/rsbinder-aidl/tests/test_enum_references.rs
+++ b/rsbinder-aidl/tests/test_enum_references.rs
@@ -528,3 +528,51 @@ pub mod CombinedFlags {
     )?;
     Ok(())
 }
+
+// Issue: https://github.com/hiking90/rsbinder/issues/68
+// Cross-package enum default values should resolve to correct module paths
+#[test]
+fn test_cross_package_enum_default_value() -> Result<(), Box<dyn Error>> {
+    let gen = rsbinder_aidl::Generator::new(false, false);
+
+    // 1. Parse and generate enum in a different package
+    //    (registers in DECLARATION_MAP + SYMBOL_TABLE)
+    let enum_doc = rsbinder_aidl::parse_document(
+        r#"
+        package test.sub;
+
+        @VintfStability
+        @Backing(type="int")
+        enum SubEnum {
+            UNKNOWN = 0,
+        }
+    "#,
+    )?;
+    let _ = gen.document(&enum_doc)?;
+
+    // 2. Parse parcelable that imports the cross-package enum
+    let parcelable_doc = rsbinder_aidl::parse_document(
+        r#"
+        package test;
+
+        import test.sub.SubEnum;
+
+        @VintfStability
+        parcelable CrossPackageParcelable {
+            SubEnum enumField = SubEnum.UNKNOWN;
+        }
+    "#,
+    )?;
+
+    // 3. Generate code and verify the path
+    let res = gen.document(&parcelable_doc)?;
+    let output = &res.1;
+
+    assert!(
+        output.contains("super::sub::SubEnum::SubEnum::UNKNOWN"),
+        "Expected cross-package path 'super::sub::SubEnum::SubEnum::UNKNOWN', got:\n{}",
+        output
+    );
+
+    Ok(())
+}

--- a/rsbinder-aidl/tests/test_interfaces.rs
+++ b/rsbinder-aidl/tests/test_interfaces.rs
@@ -79,51 +79,6 @@ pub mod ArrayOfInterfaces {
                 DEFAULT_IMPL.get_or_init(|| d).clone()
             }
         }
-        pub trait IEmptyInterfaceAsync<P>: rsbinder::Interface + Send {
-            fn descriptor() -> &'static str where Self: Sized { "ArrayOfInterfaces.IEmptyInterface" }
-        }
-        #[::async_trait::async_trait]
-        pub trait IEmptyInterfaceAsyncService: rsbinder::Interface + Send {
-            fn descriptor() -> &'static str where Self: Sized { "ArrayOfInterfaces.IEmptyInterface" }
-        }
-        impl BnEmptyInterface
-        {
-            pub fn new_async_binder<T, R>(inner: T, rt: R) -> rsbinder::Strong<dyn IEmptyInterface>
-            where
-                T: IEmptyInterfaceAsyncService + Sync + Send + 'static,
-                R: rsbinder::BinderAsyncRuntime + Send + Sync + 'static,
-            {
-                struct Wrapper<T, R> {
-                    _inner: T,
-                    _rt: R,
-                }
-                impl<T, R> rsbinder::Interface for Wrapper<T, R> where T: rsbinder::Interface, R: Send + Sync {
-                    fn as_binder(&self) -> rsbinder::SIBinder { self._inner.as_binder() }
-                    fn dump(&self, _writer: &mut dyn std::io::Write, _args: &[String]) -> rsbinder::Result<()> { self._inner.dump(_writer, _args) }
-                }
-                impl<T, R> BnEmptyInterfaceAdapter for Wrapper<T, R>
-                where
-                    T: IEmptyInterfaceAsyncService + Sync + Send + 'static,
-                    R: rsbinder::BinderAsyncRuntime + Send + Sync + 'static,
-                {
-                    fn as_sync(&self) -> &dyn IEmptyInterface {
-                        self
-                    }
-                    fn as_async(&self) -> &dyn IEmptyInterfaceAsyncService {
-                        &self._inner
-                    }
-                }
-                impl<T, R> IEmptyInterface for Wrapper<T, R>
-                where
-                    T: IEmptyInterfaceAsyncService + Sync + Send + 'static,
-                    R: rsbinder::BinderAsyncRuntime + Send + Sync + 'static,
-                {
-                }
-                let wrapped = Wrapper { _inner: inner, _rt: rt };
-                let binder = rsbinder::native::Binder::new_with_stability(BnEmptyInterface(Box::new(wrapped)), rsbinder::Stability::default());
-                rsbinder::Strong::new(Box::new(binder))
-            }
-        }
         pub trait IEmptyInterfaceDefault: Send + Sync {
         }
         pub(crate) mod transactions {
@@ -134,21 +89,13 @@ pub mod ArrayOfInterfaces {
             IEmptyInterface["ArrayOfInterfaces.IEmptyInterface"] {
                 native: {
                     BnEmptyInterface(on_transact),
-                    adapter: BnEmptyInterfaceAdapter,
-                    r#async: IEmptyInterfaceAsyncService,
                 },
                 proxy: BpEmptyInterface,
-                r#async: IEmptyInterfaceAsync,
             }
         }
         impl BpEmptyInterface {
         }
         impl IEmptyInterface for BpEmptyInterface {
-        }
-        impl<P: rsbinder::BinderAsyncPool> IEmptyInterfaceAsync<P> for BpEmptyInterface {
-        }
-        impl<P: rsbinder::BinderAsyncPool> IEmptyInterfaceAsync<P> for rsbinder::Binder<BnEmptyInterface>
-        {
         }
         impl IEmptyInterface for rsbinder::Binder<BnEmptyInterface> {
         }
@@ -171,56 +118,6 @@ pub mod ArrayOfInterfaces {
                 DEFAULT_IMPL.get_or_init(|| d).clone()
             }
         }
-        pub trait IMyInterfaceAsync<P>: rsbinder::Interface + Send {
-            fn descriptor() -> &'static str where Self: Sized { "ArrayOfInterfaces.IMyInterface" }
-            fn r#methodWithInterfaces<'a>(&'a self, _arg_iface: &'a rsbinder::Strong<dyn super::IEmptyInterface::IEmptyInterface>, _arg_nullable_iface: Option<&'a rsbinder::Strong<dyn super::IEmptyInterface::IEmptyInterface>>, _arg_iface_array_in: &'a [rsbinder::Strong<dyn super::IEmptyInterface::IEmptyInterface>], _arg_iface_array_out: &'a mut Vec<Option<rsbinder::Strong<dyn super::IEmptyInterface::IEmptyInterface>>>, _arg_iface_array_inout: &'a mut Vec<rsbinder::Strong<dyn super::IEmptyInterface::IEmptyInterface>>, _arg_nullable_iface_array_in: Option<&'a [Option<rsbinder::Strong<dyn super::IEmptyInterface::IEmptyInterface>>]>, _arg_nullable_iface_array_out: &'a mut Option<Vec<Option<rsbinder::Strong<dyn super::IEmptyInterface::IEmptyInterface>>>>, _arg_nullable_iface_array_inout: &'a mut Option<Vec<Option<rsbinder::Strong<dyn super::IEmptyInterface::IEmptyInterface>>>>) -> rsbinder::BoxFuture<'a, rsbinder::status::Result<Option<Vec<Option<String>>>>>;
-        }
-        #[::async_trait::async_trait]
-        pub trait IMyInterfaceAsyncService: rsbinder::Interface + Send {
-            fn descriptor() -> &'static str where Self: Sized { "ArrayOfInterfaces.IMyInterface" }
-            async fn r#methodWithInterfaces(&self, _arg_iface: &rsbinder::Strong<dyn super::IEmptyInterface::IEmptyInterface>, _arg_nullable_iface: Option<&rsbinder::Strong<dyn super::IEmptyInterface::IEmptyInterface>>, _arg_iface_array_in: &[rsbinder::Strong<dyn super::IEmptyInterface::IEmptyInterface>], _arg_iface_array_out: &mut Vec<Option<rsbinder::Strong<dyn super::IEmptyInterface::IEmptyInterface>>>, _arg_iface_array_inout: &mut Vec<rsbinder::Strong<dyn super::IEmptyInterface::IEmptyInterface>>, _arg_nullable_iface_array_in: Option<&[Option<rsbinder::Strong<dyn super::IEmptyInterface::IEmptyInterface>>]>, _arg_nullable_iface_array_out: &mut Option<Vec<Option<rsbinder::Strong<dyn super::IEmptyInterface::IEmptyInterface>>>>, _arg_nullable_iface_array_inout: &mut Option<Vec<Option<rsbinder::Strong<dyn super::IEmptyInterface::IEmptyInterface>>>>) -> rsbinder::status::Result<Option<Vec<Option<String>>>>;
-        }
-        impl BnMyInterface
-        {
-            pub fn new_async_binder<T, R>(inner: T, rt: R) -> rsbinder::Strong<dyn IMyInterface>
-            where
-                T: IMyInterfaceAsyncService + Sync + Send + 'static,
-                R: rsbinder::BinderAsyncRuntime + Send + Sync + 'static,
-            {
-                struct Wrapper<T, R> {
-                    _inner: T,
-                    _rt: R,
-                }
-                impl<T, R> rsbinder::Interface for Wrapper<T, R> where T: rsbinder::Interface, R: Send + Sync {
-                    fn as_binder(&self) -> rsbinder::SIBinder { self._inner.as_binder() }
-                    fn dump(&self, _writer: &mut dyn std::io::Write, _args: &[String]) -> rsbinder::Result<()> { self._inner.dump(_writer, _args) }
-                }
-                impl<T, R> BnMyInterfaceAdapter for Wrapper<T, R>
-                where
-                    T: IMyInterfaceAsyncService + Sync + Send + 'static,
-                    R: rsbinder::BinderAsyncRuntime + Send + Sync + 'static,
-                {
-                    fn as_sync(&self) -> &dyn IMyInterface {
-                        self
-                    }
-                    fn as_async(&self) -> &dyn IMyInterfaceAsyncService {
-                        &self._inner
-                    }
-                }
-                impl<T, R> IMyInterface for Wrapper<T, R>
-                where
-                    T: IMyInterfaceAsyncService + Sync + Send + 'static,
-                    R: rsbinder::BinderAsyncRuntime + Send + Sync + 'static,
-                {
-                    fn r#methodWithInterfaces(&self, _arg_iface: &rsbinder::Strong<dyn super::IEmptyInterface::IEmptyInterface>, _arg_nullable_iface: Option<&rsbinder::Strong<dyn super::IEmptyInterface::IEmptyInterface>>, _arg_iface_array_in: &[rsbinder::Strong<dyn super::IEmptyInterface::IEmptyInterface>], _arg_iface_array_out: &mut Vec<Option<rsbinder::Strong<dyn super::IEmptyInterface::IEmptyInterface>>>, _arg_iface_array_inout: &mut Vec<rsbinder::Strong<dyn super::IEmptyInterface::IEmptyInterface>>, _arg_nullable_iface_array_in: Option<&[Option<rsbinder::Strong<dyn super::IEmptyInterface::IEmptyInterface>>]>, _arg_nullable_iface_array_out: &mut Option<Vec<Option<rsbinder::Strong<dyn super::IEmptyInterface::IEmptyInterface>>>>, _arg_nullable_iface_array_inout: &mut Option<Vec<Option<rsbinder::Strong<dyn super::IEmptyInterface::IEmptyInterface>>>>) -> rsbinder::status::Result<Option<Vec<Option<String>>>> {
-                        self._rt.block_on(self._inner.r#methodWithInterfaces(_arg_iface, _arg_nullable_iface, _arg_iface_array_in, _arg_iface_array_out, _arg_iface_array_inout, _arg_nullable_iface_array_in, _arg_nullable_iface_array_out, _arg_nullable_iface_array_inout))
-                    }
-                }
-                let wrapped = Wrapper { _inner: inner, _rt: rt };
-                let binder = rsbinder::native::Binder::new_with_stability(BnMyInterface(Box::new(wrapped)), rsbinder::Stability::default());
-                rsbinder::Strong::new(Box::new(binder))
-            }
-        }
         pub trait IMyInterfaceDefault: Send + Sync {
             fn r#methodWithInterfaces(&self, _arg_iface: &rsbinder::Strong<dyn super::IEmptyInterface::IEmptyInterface>, _arg_nullable_iface: Option<&rsbinder::Strong<dyn super::IEmptyInterface::IEmptyInterface>>, _arg_iface_array_in: &[rsbinder::Strong<dyn super::IEmptyInterface::IEmptyInterface>], _arg_iface_array_out: &mut Vec<Option<rsbinder::Strong<dyn super::IEmptyInterface::IEmptyInterface>>>, _arg_iface_array_inout: &mut Vec<rsbinder::Strong<dyn super::IEmptyInterface::IEmptyInterface>>, _arg_nullable_iface_array_in: Option<&[Option<rsbinder::Strong<dyn super::IEmptyInterface::IEmptyInterface>>]>, _arg_nullable_iface_array_out: &mut Option<Vec<Option<rsbinder::Strong<dyn super::IEmptyInterface::IEmptyInterface>>>>, _arg_nullable_iface_array_inout: &mut Option<Vec<Option<rsbinder::Strong<dyn super::IEmptyInterface::IEmptyInterface>>>>) -> rsbinder::status::Result<Option<Vec<Option<String>>>> {
                 Err(rsbinder::StatusCode::UnknownTransaction.into())
@@ -235,11 +132,8 @@ pub mod ArrayOfInterfaces {
             IMyInterface["ArrayOfInterfaces.IMyInterface"] {
                 native: {
                     BnMyInterface(on_transact),
-                    adapter: BnMyInterfaceAdapter,
-                    r#async: IMyInterfaceAsyncService,
                 },
                 proxy: BpMyInterface,
-                r#async: IMyInterfaceAsync,
             }
         }
         impl BpMyInterface {
@@ -279,30 +173,9 @@ pub mod ArrayOfInterfaces {
                 self.read_response_methodWithInterfaces(_arg_iface, _arg_nullable_iface, _arg_iface_array_in, _arg_iface_array_out, _arg_iface_array_inout, _arg_nullable_iface_array_in, _arg_nullable_iface_array_out, _arg_nullable_iface_array_inout, _aidl_reply)
             }
         }
-        impl<P: rsbinder::BinderAsyncPool> IMyInterfaceAsync<P> for BpMyInterface {
-            fn r#methodWithInterfaces<'a>(&'a self, _arg_iface: &'a rsbinder::Strong<dyn super::IEmptyInterface::IEmptyInterface>, _arg_nullable_iface: Option<&'a rsbinder::Strong<dyn super::IEmptyInterface::IEmptyInterface>>, _arg_iface_array_in: &'a [rsbinder::Strong<dyn super::IEmptyInterface::IEmptyInterface>], _arg_iface_array_out: &'a mut Vec<Option<rsbinder::Strong<dyn super::IEmptyInterface::IEmptyInterface>>>, _arg_iface_array_inout: &'a mut Vec<rsbinder::Strong<dyn super::IEmptyInterface::IEmptyInterface>>, _arg_nullable_iface_array_in: Option<&'a [Option<rsbinder::Strong<dyn super::IEmptyInterface::IEmptyInterface>>]>, _arg_nullable_iface_array_out: &'a mut Option<Vec<Option<rsbinder::Strong<dyn super::IEmptyInterface::IEmptyInterface>>>>, _arg_nullable_iface_array_inout: &'a mut Option<Vec<Option<rsbinder::Strong<dyn super::IEmptyInterface::IEmptyInterface>>>>) -> rsbinder::BoxFuture<'a, rsbinder::status::Result<Option<Vec<Option<String>>>>> {
-                let _aidl_data = match self.build_parcel_methodWithInterfaces(_arg_iface, _arg_nullable_iface, _arg_iface_array_in, _arg_iface_array_out, _arg_iface_array_inout, _arg_nullable_iface_array_in, _arg_nullable_iface_array_out, _arg_nullable_iface_array_inout) {
-                    Ok(_aidl_data) => _aidl_data,
-                    Err(err) => return Box::pin(std::future::ready(Err(err.into()))),
-                };
-                let binder = self.binder.clone();
-                P::spawn(
-                    move || binder.as_proxy().unwrap().submit_transact(transactions::r#methodWithInterfaces, &_aidl_data, rsbinder::FLAG_CLEAR_BUF | rsbinder::FLAG_PRIVATE_LOCAL),
-                    move |_aidl_reply| async move {
-                        self.read_response_methodWithInterfaces(_arg_iface, _arg_nullable_iface, _arg_iface_array_in, _arg_iface_array_out, _arg_iface_array_inout, _arg_nullable_iface_array_in, _arg_nullable_iface_array_out, _arg_nullable_iface_array_inout, _aidl_reply)
-                    }
-                )
-            }
-        }
-        impl<P: rsbinder::BinderAsyncPool> IMyInterfaceAsync<P> for rsbinder::Binder<BnMyInterface>
-        {
-            fn r#methodWithInterfaces<'a>(&'a self, _arg_iface: &'a rsbinder::Strong<dyn super::IEmptyInterface::IEmptyInterface>, _arg_nullable_iface: Option<&'a rsbinder::Strong<dyn super::IEmptyInterface::IEmptyInterface>>, _arg_iface_array_in: &'a [rsbinder::Strong<dyn super::IEmptyInterface::IEmptyInterface>], _arg_iface_array_out: &'a mut Vec<Option<rsbinder::Strong<dyn super::IEmptyInterface::IEmptyInterface>>>, _arg_iface_array_inout: &'a mut Vec<rsbinder::Strong<dyn super::IEmptyInterface::IEmptyInterface>>, _arg_nullable_iface_array_in: Option<&'a [Option<rsbinder::Strong<dyn super::IEmptyInterface::IEmptyInterface>>]>, _arg_nullable_iface_array_out: &'a mut Option<Vec<Option<rsbinder::Strong<dyn super::IEmptyInterface::IEmptyInterface>>>>, _arg_nullable_iface_array_inout: &'a mut Option<Vec<Option<rsbinder::Strong<dyn super::IEmptyInterface::IEmptyInterface>>>>) -> rsbinder::BoxFuture<'a, rsbinder::status::Result<Option<Vec<Option<String>>>>> {
-                self.0.as_async().r#methodWithInterfaces(_arg_iface, _arg_nullable_iface, _arg_iface_array_in, _arg_iface_array_out, _arg_iface_array_inout, _arg_nullable_iface_array_in, _arg_nullable_iface_array_out, _arg_nullable_iface_array_inout)
-            }
-        }
         impl IMyInterface for rsbinder::Binder<BnMyInterface> {
             fn r#methodWithInterfaces(&self, _arg_iface: &rsbinder::Strong<dyn super::IEmptyInterface::IEmptyInterface>, _arg_nullable_iface: Option<&rsbinder::Strong<dyn super::IEmptyInterface::IEmptyInterface>>, _arg_iface_array_in: &[rsbinder::Strong<dyn super::IEmptyInterface::IEmptyInterface>], _arg_iface_array_out: &mut Vec<Option<rsbinder::Strong<dyn super::IEmptyInterface::IEmptyInterface>>>, _arg_iface_array_inout: &mut Vec<rsbinder::Strong<dyn super::IEmptyInterface::IEmptyInterface>>, _arg_nullable_iface_array_in: Option<&[Option<rsbinder::Strong<dyn super::IEmptyInterface::IEmptyInterface>>]>, _arg_nullable_iface_array_out: &mut Option<Vec<Option<rsbinder::Strong<dyn super::IEmptyInterface::IEmptyInterface>>>>, _arg_nullable_iface_array_inout: &mut Option<Vec<Option<rsbinder::Strong<dyn super::IEmptyInterface::IEmptyInterface>>>>) -> rsbinder::status::Result<Option<Vec<Option<String>>>> {
-                self.0.as_sync().r#methodWithInterfaces(_arg_iface, _arg_nullable_iface, _arg_iface_array_in, _arg_iface_array_out, _arg_iface_array_inout, _arg_nullable_iface_array_in, _arg_nullable_iface_array_out, _arg_nullable_iface_array_inout)
+                self.0.r#methodWithInterfaces(_arg_iface, _arg_nullable_iface, _arg_iface_array_in, _arg_iface_array_out, _arg_iface_array_inout, _arg_nullable_iface_array_in, _arg_nullable_iface_array_out, _arg_nullable_iface_array_inout)
             }
         }
         fn on_transact(

